awx/ui_next/Makefile

```diff
@@ -0,0 +1,97 @@
+## UI_NEXT_DIR: Relative path to the directory containing this Makefile
+UI_NEXT_DIR := $(patsubst %/,%,$(dir $(lastword $(MAKEFILE_LIST))))
+
+# ## Path to your local clone of the UI_NEXT repo
+# #  NOTE: you will not be able to build within the docker-compose development environment if you use this option
+UI_NEXT_LOCAL ?=
+
+# Git repo and branch to the UI_NEXT repo
+UI_NEXT_GIT_REPO ?= https://github.com/ansible/ansible-ui.git
+UI_NEXT_GIT_BRANCH ?= main
+
+.PHONY: ui-next
+## Default build target of ui-next Makefile, builds ui-next/build
+ui-next: ui-next/build
+
+.PHONY: ui-next/build
+## Build ui-next/build
+ui-next/build: $(UI_NEXT_DIR)/build
+
+## Real target for ui-next.
+$(UI_NEXT_DIR)/build: ui-next/src/build
```

Comment:
Somewhat in line with John's comment: It's not always clear which of these targets are meant to be PHONY and which are meant to be real targets. And, more importantly to functionality:
1. Real targets should probably never depend on phony targets because PHONY targets will always evaluate as needing updating, and so they'll make the "real" targets always need updating, too, since the real targets *depend* on the always-out-of-date PHONY target, and if something might have changed with it, then something might have changed with our target and...
2. Real directories should almost never be real targets
Similar, but distinct reasoning: A POSIX directory is "updated" or "changed" anytime any metadata about it changes (!), so that would include any files atime (on certain filesystems), new/removed directory, etc. etc. and GNU Make looks at that as a trigger to update the target, so directories will often need updating.
It's not immediately super clear to me what the intent is, so I'm not sure what to recommend instead, but it's late on a Friday. I wouldn't block merging this if it works, but would definitely want to rewrite a lot of it in a followup commit.

Reply:
explicitly PHONY the make targets